<a href="https://colab.research.google.com/github/kinoi-r/imersao_dados_python/blob/main/Aula_0_de_Python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [25]:
import pandas as pd

Há muitas ferramentas instaladas no Collab, para acessar basta importar. Aqui importamos o pandas e a apelidamos de "pd" como uma forma mais rápida de se referir a biblioteca, usando "as" que, em inglês, significa "como".
Pandas é uma biblioteca de análise de dados e ao importá-la, todas as suas funções já estão disponíveis para serem utilizadas.

No Python, usando tanto essa IDE quanto outras, é importante separar o código por células, ao invés de escrever tudo em uma única célula. Apesar de funcionar se escrevermos tudo em uma única célula, é muito mais legível se escrevermos uma instrução por célula, tornando o código mais limpo e organizado, e mais fácil de identificar erros.

In [26]:
df = pd.read_csv("https://raw.githubusercontent.com/guilhermeonrails/data-jobs/refs/heads/main/salaries.csv")

"df" foi o nome escolhido pra identificar o arquivo dentro do Python onde estará a base de dados, que significa "data frame". Pode dar o nome que quiser, mas "df" é uma prática comum. Nessa célula, estamos dizendo que Python deve ler uma base de dados "df" usando o Pandas "pd" e o link é onde está essa base.

In [28]:
df.head(10)

work_year experience_level employment_type           job_title  salary  \
0     2025.0               SE              FT  Solutions Engineer  214000   
1     2025.0               SE              FT  Solutions Engineer  136000   
2     2025.0               MI              FT       Data Engineer  158800   
3     2025.0               MI              FT       Data Engineer  139200   
4     2025.0               EN              FT       Data Engineer   90000   
5     2025.0               EN              FT       Data Engineer   80000   
6     2025.0               SE              FT      Data Scientist  185000   
7     2025.0               SE              FT      Data Scientist  148000   
8     2025.0               MI              FT      Data Scientist  135000   
9     2025.0               MI              FT      Data Scientist  108000   

  salary_currency  salary_in_usd employee_residence  remote_ratio  \
0             USD         214000                 US           100   
1             USD         136000                 US           100   
2             USD         158800                 AU             0   
3             USD         139200                 AU             0   
4             USD          90000                 US             0   
5             USD          80000                 US             0   
6             USD         185000                 US             0   
7             USD         148000                 US             0   
8             USD         135000                 CA             0   
9             USD         108000                 CA             0   

  company_location company_size  
0               US            M  
1               US            M  
2               AU            M  
3               AU            M  
4               US            M  
5               US            M  
6               US            M  
7               US            M  
8               CA            M  
9               CA            M

Nessa tabela, ao colocarmos df.head() com parênteses vazios, o Python, a partir do número 0, vai enumerar até o numero 4 as colunas da tabela, mas se adicionar uma quantidade dentro dos parênteses, o Python vai ordenar em uma quantidade maior de colunas definidas, como df.head(10)

In [29]:
df.info

<bound method DataFrame.info of         work_year experience_level employment_type                 job_title  \
0          2025.0               SE              FT        Solutions Engineer   
1          2025.0               SE              FT        Solutions Engineer   
2          2025.0               MI              FT             Data Engineer   
3          2025.0               MI              FT             Data Engineer   
4          2025.0               EN              FT             Data Engineer   
...           ...              ...             ...                       ...   
133344     2020.0               SE              FT            Data Scientist   
133345     2021.0               MI              FT  Principal Data Scientist   
133346     2020.0               EN              FT            Data Scientist   
133347     2020.0               EN              CT     Business Data Analyst   
133348     2021.0               SE              FT            Data Scientist   

         salary salary_currency  salary_in_usd employee_residence  \
0        214000             USD         214000                 US   
1        136000             USD         136000                 US   
2        158800             USD         158800                 AU   
3        139200             USD         139200                 AU   
4         90000             USD          90000                 US   
...         ...             ...            ...                ...   
133344   412000             USD         412000                 US   
133345   151000             USD         151000                 US   
133346   105000             USD         105000                 US   
133347   100000             USD         100000                 US   
133348  7000000             INR          94665                 IN   

        remote_ratio company_location company_size  
0                100               US            M  
1                100               US            M  
2                  0               AU            M  
3                  0               AU            M  
4                  0               US            M  
...              ...              ...          ...  
133344           100               US            L  
133345           100               US            L  
133346           100               US            S  
133347           100               US            L  
133348            50               IN            L  

[133349 rows x 11 columns]>

Ao usarmos df.info, o Python nos mostra a tabela em outro formato. A tabela também mostra vários tipos de dados e cada um possuí um formato. Algumas tabelas são extremamente grandes, ultrapassando mil colunas, e o .info acaba nos ajudando a ter uma visualização melhor.

In [30]:
df.describe

<bound method NDFrame.describe of         work_year experience_level employment_type                 job_title  \
0          2025.0               SE              FT        Solutions Engineer   
1          2025.0               SE              FT        Solutions Engineer   
2          2025.0               MI              FT             Data Engineer   
3          2025.0               MI              FT             Data Engineer   
4          2025.0               EN              FT             Data Engineer   
...           ...              ...             ...                       ...   
133344     2020.0               SE              FT            Data Scientist   
133345     2021.0               MI              FT  Principal Data Scientist   
133346     2020.0               EN              FT            Data Scientist   
133347     2020.0               EN              CT     Business Data Analyst   
133348     2021.0               SE              FT            Data Scientist   

         salary salary_currency  salary_in_usd employee_residence  \
0        214000             USD         214000                 US   
1        136000             USD         136000                 US   
2        158800             USD         158800                 AU   
3        139200             USD         139200                 AU   
4         90000             USD          90000                 US   
...         ...             ...            ...                ...   
133344   412000             USD         412000                 US   
133345   151000             USD         151000                 US   
133346   105000             USD         105000                 US   
133347   100000             USD         100000                 US   
133348  7000000             INR          94665                 IN   

        remote_ratio company_location company_size  
0                100               US            M  
1                100               US            M  
2                  0               AU            M  
3                  0               AU            M  
4                  0               US            M  
...              ...              ...          ...  
133344           100               US            L  
133345           100               US            L  
133346           100               US            S  
133347           100               US            L  
133348            50               IN            L  

[133349 rows x 11 columns]>

df.describe nos permite aprofundar na tabela pra entender melhor e algo que podemos destacar é que Python pegou apenas as colunas que são numericas, então tudo que era do tipo float e do tipo int está nessa tabela .describe, que traz umas estatísticas que chamamos de descritivas.

In [31]:
df.shape

(133349, 11)

df.shape não utiliza parênteses, pois é um atributo, não um método. Ele traz a dimensão do arquivo. Nesse caso, uma tupla, já que há dois valores. O número 133.349 linhas e 11 colunas. É importante saber essa informação porque podemos precisar realizar operações matemáticas dessas matrizes.

In [32]:
linhas, colunas = df.shape[0], df.shape[1]
print('Linhas:', linhas)
print('Colunas:', colunas)

Linhas: 133349
Colunas: 11


Uma maneira mais organizada pra saída de dados.

In [39]:
df.columns

Index(['ano', 'nivel_experiencia', 'tipo_emprego', 'titulo_cargo', 'salario',
       'moeda_salario', 'salario_em_usd', 'residencia_funcionario',
       'proporcao_remoto', 'localizacao_empresa', 'tamanho_empresa'],
      dtype='object')

df.columns serve pra exibir de forma organizada apenas o nome das colunas.

Como a Colab é linkada com a Gemini, a inteligência artificial da Google, podemos acessar Generate with AI e solicitar que traduza o nome das colunas pra português. Não é necessário copiar o nome das colunas, basta dizer "traduza para mim o nome das colunas dataframe df" que já é suficiente. Ela vai gerar um código pra conferirmos, se estiver tudo certo, basta rodar e ele renomeará as colunas com os nomes em português.


DICA: Em um arquivo REAL, devemos realizar renomeações unicamente em novos arquivos, para não perdermos os arquivos originais.

In [40]:
df.rename(columns={
    'ano': 'ano',
    'nivel_experiencia': 'senioridade',
    'tipo_emprego': 'contrato',
    'titulo_cargo': 'cargo',
    'salario': 'salario',
    'moeda_salario': 'moeda',
    'salario_em_usd': 'usd',
    'residencia_funcionario': 'residencia',
    'proporcao_remoto': 'remoto',
    'localizacao_empresa': 'empresa',
    'tamanho_empresa': 'tamanho_empresa'
}, inplace=True)


print("Novos nomes das colunas:")
print(df.columns)

Novos nomes das colunas:
Index(['ano', 'senioridade', 'contrato', 'cargo', 'salario', 'moeda', 'usd',
       'residencia', 'remoto', 'empresa', 'tamanho_empresa'],
      dtype='object')


In [41]:
df['senioridade'].value_counts()

senioridade
SE    77241
MI    40465
EN    12443
EX     3200
Name: count, dtype: int64

Agora utilizamos df['nivel_experiencia'] com o método value_counts() pra chamar o arquivo df e revelar a frequência, contar os valores de cada categoria SE (profissional sênior), MI (profissional mid, intermediário), EN (profissional junior, iniciante) e EX (executivo, liderança) dentro da coluna de nível de experiência.

In [43]:
df['contrato'].value_counts()

contrato
FT    132563
CT       394
PT       376
FL        16
Name: count, dtype: int64

Podemos repetir com outras colunas. Nesse resultado, vemos que na categoria FT (Full time, integral), há 132563 funcionários. CT (Contrato temporário), há 394 funcionários. PT (Part time, meio período), há 376 funcionários. FL (freelance) há 16 funcionários.

In [44]:
df['remoto'].value_counts()

remoto
0      105312
100     27718
50        319
Name: count, dtype: int64

Repetindo com "remoto" pra ter melhor visualização dos contratos remotos da empresa. Em "0" indica funcionários presenciais, em "100" indica funcionários remotos e em "50" indica vagas híbridas.

In [45]:
df['tamanho_empresa'].value_counts()

tamanho_empresa
M    129561
L      3574
S       214
Name: count, dtype: int64

Aqui "M" indica empresas médias, que são em maior número, de acordo com o resultado. "L" indica empresas grandes e "S" empresas pequenas.

Note como há uma grande quantidade de siglas aparecendo. O ideal é deixar tudo com nomes intuitivos e facéis de entender, especialmente se houver uma apresentação desse trabalho há alguém interessado. Portanto, podemos usar novamente a função Generate with AI pra renomear o nome das categorias das colunas apresentadas.

In [46]:
df['senioridade'] = df['senioridade'].replace({
    'SE': 'Senior',
    'MI': 'Pleno',
    'EN': 'Junior',
    'EX': 'Executivo'
})

print("Novas categorias na coluna 'senioridade':")
print(df['senioridade'].value_counts())

Novas categorias na coluna 'senioridade':
senioridade
Senior       77241
Pleno        40465
Junior       12443
Executivo     3200
Name: count, dtype: int64


In [47]:
df['contrato'] = df['contrato'].replace({
    'FT': 'Integral',
    'CT': 'Temporário',
    'PT': 'Meio Período',
    'FL': 'Freelance'
})

print("Novas categorias na coluna 'contrato':")
print(df['contrato'].value_counts())

Novas categorias na coluna 'contrato':
contrato
Integral        132563
Temporário         394
Meio Período       376
Freelance           16
Name: count, dtype: int64


In [48]:
df['tamanho_empresa'] = df['tamanho_empresa'].replace({
    'M': 'Média',
    'L': 'Grande',
    'S': 'Pequena'
})

print("Novas categorias na coluna 'tamanho_empresa':")
print(df['tamanho_empresa'].value_counts())

Novas categorias na coluna 'tamanho_empresa':
tamanho_empresa
Média      129561
Grande       3574
Pequena       214
Name: count, dtype: int64


In [49]:
df['remoto'] = df['remoto'].replace({
    0: 'Presencial',
    100: 'Remoto',
    50: 'Híbrido'
})

print("Novas categorias na coluna 'remoto':")
print(df['remoto'].value_counts())

Novas categorias na coluna 'remoto':
remoto
Presencial    105312
Remoto         27718
Híbrido          319
Name: count, dtype: int64


In [50]:
df.head()

ano senioridade  contrato               cargo  salario moeda     usd  \
0  2025.0      Senior  Integral  Solutions Engineer   214000   USD  214000   
1  2025.0      Senior  Integral  Solutions Engineer   136000   USD  136000   
2  2025.0       Pleno  Integral       Data Engineer   158800   USD  158800   
3  2025.0       Pleno  Integral       Data Engineer   139200   USD  139200   
4  2025.0      Junior  Integral       Data Engineer    90000   USD   90000   

  residencia      remoto empresa tamanho_empresa  
0         US      Remoto      US           Média  
1         US      Remoto      US           Média  
2         AU  Presencial      AU           Média  
3         AU  Presencial      AU           Média  
4         US  Presencial      US           Média

Após todas as renomeações nas siglas originais em inglês, podemos visualizar a tabela traduzida pra português e mais fácil de entender.

O uso do .describe() estende-se aos dados categóricos, permitindo uma análise rápida de frequências e valores únicos. Essa funcionalidade é fundamental para obter insights estratégicos além do universo numérico

In [51]:
df.describe(include='object')

senioridade  contrato           cargo   moeda residencia      remoto  \
count       133349    133349          133349  133349     133349      133349   
unique           4         4             390      26        102           3   
top         Senior  Integral  Data Scientist     USD         US  Presencial   
freq         77241    132563           17314  126140     119579      105312   

       empresa tamanho_empresa  
count   133349          133349  
unique      95               3  
top         US           Média  
freq    119641          129561

In [52]:
df.describe()

ano       salario            usd
count  133339.000000  1.333490e+05  133349.000000
mean     2024.358770  1.632833e+05  157617.272098
std         0.680627  2.173860e+05   74288.363097
min      2020.000000  1.400000e+04   15000.000000
25%      2024.000000  1.060200e+05  106000.000000
50%      2024.000000  1.470000e+05  146206.000000
75%      2025.000000  1.990000e+05  198000.000000
max      2025.000000  3.040000e+07  800000.000000